backend/systematic_indentation_fix.py

```
@@ -0,0 +1,211 @@
+            j = i + 1
+            while j < len(lines):
+                next_line = lines[j]
+                if next_line.strip():
+                    next_indent = len(next_line) - len(next_line.lstrip())
+                    if next_indent <= try_indent:
+                        # We've exited the try block
+                        if next_line.strip().startswith('except') or next_line.strip().startswith('finally'):
+                            has_except = True
+                        break
+                    elif 'except' in next_line or 'finally' in next_line:
+                        has_except = True
```

Ruff (0.12.2)

Local variable `has_except` is assigned to but never used

Remove assignment to unused variable `has_except`

(F841)